src/qibosoq/rfsoc_server.py

```diff
@@ -42,12 +50,26 @@ def execute_program(self, data: dict) -> dict:
             (dict): dictionary with two keys (i, q) to lists of values
         """
         if data["operation_code"] == "execute_pulse_sequence":
-            program = ExecutePulseSequence(global_soc, data["cfg"], data["sequence"], data["qubits"])
-        elif data["operation_code"] == "execute_single_sweep":
-            program = ExecuteSingleSweep(global_soc, data["cfg"], data["sequence"], data["qubits"], data["sweeper"])
+            program = ExecutePulseSequence(
+                global_soc,
+                Config(**data["cfg"]),
+                [Pulse(**pulse) for pulse in data["sequence"]],
+                [Qubit(**qubit) for qubit in data["qubits"]],
+            )
+        elif data["operation_code"] == "execute_sweeps":
+            program = ExecuteSweeps(
+                global_soc,
+                Config(**data["cfg"]),
+                [Pulse(**pulse) for pulse in data["sequence"]],
+                [Qubit(**qubit) for qubit in data["qubits"]],
```

Comment:
These lines look exactly like those above :)

Reply:
But what can i do? I would like to do something similar to 
```python
if sequence:
    program = ExecutePulseSequence
elif sweepers:
    program = ExecuteSweeps

program = program ( soc, Config(**data["cfg"], ...)
```
but in the case of sweepers, the program takes one variable more and I do not know what to do

Reply:
There are many solutions, the complicate part is just to find the simplest one.

A possible one can be:
```py
if sequence:
    progcls = ExecutePulseSequence
    args = ()
elif sweepers:
    progcls = ExecuteSweeps
    args = ([Sweeper(**sweeper) for sweeper in data["sweepers"]],)

program = progcls(soc, Config(**data["cfg"]), ..., *args)
```
but it is not necessarily the best one.